love_letter_sim2.py

```diff
@@ -183,31 +201,92 @@ def check_hand(self):
                 for c in self.hand:
                     if isinstance(c, King) or isinstance(c, Prince):
                         print("have to discard and end turn")
+                        # JHA - we have to do something different here.
                         self.hand.remove(card)
 
-    def play_card(card_name, target_name=""):
-        pass
-        # I think playing a card is a player action that invokes
+    def play_card(self):
+        # MA - I think playing a card is a player action that invokes
         # a card object. Many cards need a target but not all
         # I'm guessing I could do it better than the empty string
         # I have above but that's what I have now
+        # JHA - I think the player object should be the one deciding which
+        # card to play.  Therefore shouldn't need the params
+
+        # for now, just removing first card
+        played_card = self.hand[0]
+        print("{0} played {1}".format(self.id, played_card))
+        self.hand.pop(0)
 
     def compare_card(target):
         pass
         # discard Baron
         # compare cards and determine who is eliminated
         # eliminate player
 
+    def __str__(self):
+        if len(self.hand) == 1:
+            return "{0} holds {1}".format(self.id, self.hand[0])
+        else:
+            return "{0} holds {1} and {2}".format(self.id, self.hand[0],
+                                                  self.hand[1])
+
 
 class Game:
-    def __init__(self, players):
-        self.over = False
-        self.start_player = random.choice(players)
-        # This varies turn order and makes it easier to loop through
-        # the players sequentially. I still have it in the back of
-        # my head that I eventually will add multiplayer support
-        random.shuffle(players)
-        self.turn_order = itertools.cycle(players)
+    def __init__(self, *args):
+        # initialze deck
+        self.deck = Deck()
+
+        # initialize players
+        self.players = list()
+        for arg in args:
+            player = Player(arg)
+            player.draw_a_card(self.deck)
+            self.players.append(player)
+        self.start_player = random.choice(self.players)
```

Comment:
Don't need this line. This was a line I put in before but realized, it doesn't matter as the shuffle then cycle determines this. Not sure how to do a line item veto in Git.